### PR TITLE
Add toggle for always placing the dragon egg

### PR DESCRIPTION
**Pull request by HexedHero**

### CODE REVIEW

File: Spigot-Server-Patches/0602-Add-toggle-for-always-placing-the-dragon-egg.patch [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+index 38dc6086d18951e065d4048d1d8eee288c5c5fd1..7f11adbc5d8527b50f7657d46459516a471017b3 100644
+--- a/src/main/java/net/minecraft/server/EnderDragonBattle.java
++++ b/src/main/java/net/minecraft/server/EnderDragonBattle.java
+@@ -358,7 +358,7 @@ public class EnderDragonBattle {
+             this.bossBattle.setVisible(false);
+             this.generateExitPortal(true);
+             this.n();
+-            if (!this.previouslyKilled) {
++            if (this.world.paperConfig.enderDragonsDeathAlwaysPlacesDragonEgg || !this.previouslyKilled) { // Paper - always place dragon egg
+                 this.world.setTypeUpdate(this.world.getHighestBlockYAt(HeightMap.Type.MOTION_BLOCKING, WorldGenEndTrophy.a), Blocks.DRAGON_EGG.getBlockData());
```
Comment (Octol1ttle) [inline Spigot-Server-Patches/0602-Add-toggle-for-always-placing-the-dragon-egg.patch]: Can you add obfhelpers please?

Comment (electronicboy) [reply Spigot-Server-Patches/0602-Add-toggle-for-always-placing-the-dragon-egg.patch]: Not needed here...

Comment (Octol1ttle) [reply Spigot-Server-Patches/0602-Add-toggle-for-always-placing-the-dragon-egg.patch]: Sorry

**Review by Proximyst — APPROVED** [review]

**Comment by pop4959** [comment]
What's funny is I have a commit in one of my private plugins for this. On a multiplayer survival server it's definitely a good QOL feature, since players won't feel that they've "lost out" on the trophy from the dragon.

**Comment by BillyGalbreath** [comment]
You're killing me here... https://github.com/pl3xgaming/Purpur/blob/ver/1.16.4/patches/server/0048-Ender-dragon-always-drop-egg.patch ^_^

Might as well steal this one, too. https://github.com/pl3xgaming/Purpur/blob/ver/1.16.4/patches/server/0049-Ender-dragon-always-drop-full-exp.patch

**Comment by MiniDigger** [comment]
Went ahead and rebased + fixed the patch author.

@HexedHero, if you PR patches you didn't originally write (which in itself is fine and appreciated!), please follow the licence of the patches, don't change the author of the patches (if porting the patch was non trivial, you can add yourself as a coauthor to get attribution for the patch) and if possible, link to the original patch.

Your commit to add the patch will be still be under your name, so it gets properly recorded as a contribution to paper on github, if thats what you care about.

**Pull request merged**